packages/material-ui/src/Select/SelectInput.js

```diff
@@ -141,6 +150,22 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     event.preventDefault();
     displayRef.current.focus();
 
+    const doc = ownerDocument(event.target);
+    if (handleClick.current) {
+      doc.removeEventListener('click', handleClick.current, true);
+    }
+
+    // Since the targets of the `mousedown` and `mouseup` events are different,
```

Comment:
It's not clear at this point in the code that this is the case. Wouldn't you want to check that during mouseup?

Reply:
To check during `mouseup` I would have to add another listener to the document and from this listener add another one to cancel and re-dispatch the `click`.